File: src/main/java/com/viniciusfinger/rsflood/model/Address.java
package com.viniciusfinger.rsflood.model;

import java.util.List;

import com.viniciusfinger.rsflood.enums.HomeZone;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String street;

    private String number;

    private String cep;

    private String neighbourhood;

    private String city;

    @Enumerated(EnumType.STRING)
    private HomeZone zone;

    private String observation;

    @OneToMany(mappedBy = "address")
    private List<Home> homes;

}

File: frontend/src/app/model/address.ts
import { HomeZone } from "./enum/home-zone";
import { Home as Home } from "./home";

export class Address {
    id: number | null;
    street: string;
    number: string;
    cep: string;
    neighbourhood: string;
    city: string;
    zone: HomeZone;
    observation: string;
    homes: Home[];

    constructor(
        id: number | null,
        street: string,
        number: string,
        cep: string,
        neighbourhood: string,
        city: string,
        zone: HomeZone,
        observation: string,
        homes: Home[]
    ) {
        this.id = id;
        this.street = street;
        this.number = number;
        this.cep = cep;
        this.neighbourhood = neighbourhood;
        this.city = city;
        this.zone = zone;
        this.observation = observation;
        this.homes = homes;
    }
}
